### PR TITLE
Add textDecoration prop support to Text

### PR DESCRIPTION
**Pull request by pvenable**

### CODE REVIEW

File: src/Text/__tests__/Text.test.tsx [hunk 1]
```diff
@@ -9,4 +9,14 @@ describe('<Text />', () => {
...
   });
+
+  it('supports textDecoration prop', () => {
+    const tree = createWithTheme(
+      <Text textDecoration="line-through">
+        The quick brown fox jumps over the lazy dog
+      </Text>
+    );
+
+    expect(tree).toMatchSnapshot();
```
Comment (pvenable) [inline src/Text/__tests__/Text.test.tsx:20]: Big diff, so... here's the test.

File: src/utils/textDecoration.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  textDecoration?:
+    | ResponsiveValue<CSS.Property.TextDecoration, ThemeType>
+    | undefined;
+};
+
+const textDecoration = style({
+  prop: 'textDecoration'
+});
+
+export default textDecoration;
```
Comment (pvenable) [inline src/utils/textDecoration.ts:14]: Here's the prop implementation.

File: src/Text/Text.tsx [hunk 3]
```diff
@@ -40,6 +41,7 @@ export type Props = Omit<HTMLAttributes<HTMLElement>, 'color'> &
   OverflowProps &
   SpaceProps &
   TextAlignProps &
+  TextDecorationProps &
```
Comment (pvenable) [inline src/Text/Text.tsx:44]: And the usage here and below.

File: package.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
     "build-storybook": "build-storybook",
     "lint": "yarn eslint && stylelint src",
     "eslint": "eslint src stories .storybook",
-    "test:ci": "yarn test && yarn lint",
+    "test:ci": "yarn test --runInBand && yarn lint",
```
Comment (pvenable) [inline package.json:13]: I've been seeing a number of inexplicable timeouts on CI lately, so trying this (https://jestjs.io/docs/cli#--runinband) as is often recommended for low-resource CI runs.  We'll see whether it ends up helping, but it shouldn't hurt.

**Review by xxx — APPROVED** [review]

**Pull request merged**